nemo/collections/common/data/lhotse/text_adapters.py

```diff
@@ -95,3 +116,416 @@ def __iter__(self) -> Iterator[TextPairExample]:
                     if self.target_language is not None:
                         example.target.language = self.target_language
                     yield example
+
+
+@dataclass
+class NeMoSFTExample:
+    data: dict
+    language: str | None = None
+    input_ids: np.ndarray | None = None
+    context_ids: np.ndarray | None = None
+    answer_ids: np.ndarray | None = None
+    mask: np.ndarray | None = None
+    metadata: dict | None = None
+
+    def tokenize(self, tokenizer: TokenizerWrapper | TokenizerSpec) -> "NeMoSFTExample":
+        """
+        Create a tokenized variant of this example given a tokenizer (i.e. fill the optional fields).
+        Supports BPE tokenizers and aggregate tokenizers.
+
+        The tokenization is compatible with Megatron's :class:`GPTSFTChatDataset`.
+        """
+        special_tokens = {
+            "system_turn_start": "<extra_id_0>",
+            "turn_start": "<extra_id_1>",
+            "label_start": "<extra_id_2>",
+            "end_of_turn": "\n",
+            "end_of_name": "\n",
+        }
```

Comment:
The special tokens are tricky: perhaps we need to ensure that they are included in the tokenizer's vocab.